pkg/api/v1alpha1/users.go

```diff
@@ -33,7 +34,41 @@ const (
 	UserStatusSuspended = "suspended"
 )
 
-var permittedListUsersParams = []string{"external_id", "email", "metadata"}
+var (
+	permittedListUsersParams = []string{"external_id", "email", "metadata"}
+	metadataKeyPattern       = regexp.MustCompile(`^[a-zA-Z0-9_/]+$`)
```

Comment:
Should we also allow `-` ?

Also wonder if we should require that the first character be alpha(num)?, not sure if we'd want a key to just be `_` or `/`